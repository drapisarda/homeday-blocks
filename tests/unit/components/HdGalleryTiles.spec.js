import { mount } from '@vue/test-utils';
import HdGalleryTiles from '@/components/gallery/HdGalleryTiles.vue';
import ITEMS from '@/stories/mocks/GALLERY_ITEMS';

describe('HdGalleryTiles', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(HdGalleryTiles, {
      propsData: {
        items: ITEMS,
      },
    });
  });

  it('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits the proper event when the user clicks on an image', () => {
    const chosenImageIndex = 2;
    const image = wrapper.findAll('.gallery-tile').at(chosenImageIndex);
    image.trigger('click');

    expect(wrapper.emitted('itemClick')[0][0]).toBe(chosenImageIndex);
  });

  it('emits the proper event clicking on the _more photos_ button', () => {
    const button = wrapper.findAll('.gallery-tile__more-photos');
    button.trigger('click');

    expect(wrapper.emitted('itemClick')[0][0]).toBe(0);
  });
});
