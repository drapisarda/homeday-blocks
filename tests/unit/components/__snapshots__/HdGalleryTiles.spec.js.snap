// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdGalleryTiles renders component 1`] = `
<div class="gallery-tiles">
  <section class="gallery-tiles__container">
    <div tabindex="0" class="gallery-tile">
      <div class="lazy-image gallery-tile__image" style="background-image: url(https://dummyimage.com/154x86?text=1+thumb+TL+to+D);"></div>
    </div>
    <div tabindex="0" class="gallery-tile">
      <div class="lazy-image gallery-tile__image" style="background-image: url(https://dummyimage.com/154x86?text=2+thumb+TL+to+D);"></div>
    </div>
    <div tabindex="0" class="gallery-tile">
      <div class="lazy-image gallery-tile__image" style="background-image: url(https://dummyimage.com/154x86?text=3+thumb+default+density);"></div>
    </div>
    <div tabindex="0" class="gallery-tile">
      <div class="lazy-image gallery-tile__image" style="background-image: url(https://dummyimage.com/306x172?text=4+thumb+default+density);"></div>
    </div>
    <div tabindex="0" class="gallery-tile">
      <div class="lazy-image gallery-tile__image" style="background-image: url(https://dummyimage.com/306x172?text=6);"></div>
    </div>
  </section> <button class="btn gallery-tile__more-photos">Weitere Fotos</button>
</div>
`;
